azurelinuxagent/common/event.py

```diff
@@ -446,11 +445,11 @@ def add_event(self, name, op=WALAEventOperation.Unknown, is_success=True, durati
 
     def add_log_event(self, level, message):
         event = TelemetryEvent(TELEMETRY_LOG_EVENT_ID, TELEMETRY_LOG_PROVIDER_ID)
-        event.parameters.append(TelemetryEventParam('EventName', WALAEventOperation.Log))
-        event.parameters.append(TelemetryEventParam('CapabilityUsed', logger.LogLevel.STRINGS[level]))
-        event.parameters.append(TelemetryEventParam('Context1', self._clean_up_message(message)))
-        event.parameters.append(TelemetryEventParam('Context2', ''))
-        event.parameters.append(TelemetryEventParam('Context3', ''))
+        event.parameters.append(TelemetryEventParam(GuestAgentGenericLogsSchema.EventName, WALAEventOperation.Log))
+        event.parameters.append(TelemetryEventParam(GuestAgentGenericLogsSchema.CapabilityUsed, logger.LogLevel.STRINGS[level]))
+        event.parameters.append(TelemetryEventParam(GuestAgentGenericLogsSchema.Context1, self._clean_up_message(message)))
+        event.parameters.append(TelemetryEventParam(GuestAgentGenericLogsSchema.Context2, datetime.utcnow().strftime(logger.Logger.LogTimeFormatInUTC)))
+        event.parameters.append(TelemetryEventParam(GuestAgentGenericLogsSchema.Context3, ''))
```

Comment:
Would you want to use this field for the timestamp of when the event is sent? Asking since Context2 now contains the timestamp when the event is created.

Reply:
Well logically it should always have the timestamp of when the event was created right? The main reason why I added timestamp here is to keep it in sync with the current Windows agent behavior. Also it would be easier for publishers if we have consistency in our schema to make it easy for them to setup their queries.

Reply:
My comment was more about potentially adding the timestamp of sending the event to Context3 and if there were plans for that (does Windows have it?). I totally agree the timestamp of event creation is important.

Reply:
Ohh sorry misunderstood that. Windows uses Context3 as the Name equivalent of who created the event. 

Eg: Context3
Microsoft-WindowsAzure-Diagnostics/GuestAgent
Microsoft-WindowsAzure-Diagnostics/Heartbeat
Microsoft-WindowsAzure-Diagnostics/GuestAgent

We can do something similar. We can maybe repurpose the OpCode name for the time that we send the event for GenericLogs table maybe? But that's outside the scope of this PR

Reply:
Gotcha.